### PR TITLE
Local api loader improvements

### PR DESCRIPTION
**Pull request by myangelkamikaze**
Older data might not work totally as expected, because it's using the current ship/equipment data trying to simulate what happened during the sortie, meaning if you don't have the exact ship/equipment anymore, it can break stuff.
The missing data can be reconstructed, but it's a lot of work and technically isn't needed for the current use case, which is mainly for #484 testing.

Data from before the opening torpedo rework won't work either till #484 is merged. That should come after this PR is merged.

### CODE REVIEW

File: ElectronicObserver/Data/KCDatabase.cs [hunk 1]
```diff
@@ -206,8 +207,53 @@ private KCDatabase()
 		TsunDbSubmission = new TsunDbSubmissionManager();
 		FleetPreset = new FleetPresetManager();
 		Translation = new DataAndTranslationManager();
+
+#if DEBUG
+		// data needed for loading old event battles via local api loader
+		// the values don't really matter, they just need to exist
```
Comment (Jebzou) [inline ElectronicObserver/Data/KCDatabase.cs:213]: woudn't it be easier to maintain to programmatically call AddWorld with something like for i = 56, while i < last event id; i++
and call AddMap 7 times for each event

Comment (myangelkamikaze) [reply ElectronicObserver/Data/KCDatabase.cs:213]: true

File: ElectronicObserver/Window/Dialog/DialogLocalAPILoader2.cs [hunk 2]
```diff
@@ -211,15 +283,85 @@ private void ExecuteAPI(string filename)
...
 	private void APICaller_DoWork(object sender, DoWorkEventArgs e)
 	{
+		if (e.Argument is not IEnumerable<string?> files) return;
 
-		var files = e.Argument as IOrderedEnumerable<string>;
-		var act = new Action<string>(ExecuteAPI);
+		Action<string> act = SortieRecord switch
+		{
+			null => ExecuteAPI,
+			_ => ExecuteSortieRecordApi
```
Comment (Jebzou) [inline ElectronicObserver/Window/Dialog/DialogLocalAPILoader2.cs:361]: missing trailing comma

File: ElectronicObserver/Window/Dialog/DialogLocalAPILoader2.cs [hunk 3]
```diff
@@ -211,15 +283,85 @@ private void ExecuteAPI(string filename)
...
+						},
+					})
+					.OfType<FleetDataDto>()
+					.ToList();
+
+				apiFile.Content = JsonSerializer.Serialize(apiFilePortResponse, JsonSerializerOptions);
+			}
+		}
+
+		switch (apiFileType)
```
Comment (Jebzou) [inline ElectronicObserver/Window/Dialog/DialogLocalAPILoader2.cs:337]: maybe add default case, no idea how it behaves without it when running into something that isn't request or response (which shouldn't happen of course)

File: ElectronicObserver/Window/Dialog/DialogLocalAPILoader2.cs [hunk 4]
```diff
@@ -387,4 +529,18 @@ private void APIView_CellMouseDoubleClick(object sender, DataGridViewCellMouseEv
 			Utility.Logger.Add(1, string.Format(LocalAPILoader2Resources.FailedToStart, ex.GetType().Name, ex.Message));
 		}
 	}
+
+	private string? LoadApiResponse(string apiName)
+	{
+		Configuration.ConfigurationData Config = Configuration.Config;
```
Comment (Jebzou) [inline ElectronicObserver/Window/Dialog/DialogLocalAPILoader2.cs:535]: Config => config

Comment (myangelkamikaze) [reply ElectronicObserver/Window/Dialog/DialogLocalAPILoader2.cs]: good old copy-paste

**Review by Jebzou — APPROVED** [review]

**Pull request merged**